gammapy/datasets/fermi.py

```diff
@@ -215,6 +217,147 @@ def fetch_fermi_diffuse_background_model(filename='gll_iem_v02.fit'):
+
+            flux = self.cat_row[self.y_labels[i]]
+            
+            # Require both a detection and a lower bound
+            if np.isnan(flux) == False:
+
+                y_err_label = "Unc_" + self.y_labels[i]
+                flux_lower_bound = self.cat_row[y_err_label][0]
+                if np.isnan(flux_lower_bound) == False:
+                    valid_indices.append(i)
+
+        y_vals = np.array([self.cat_row[i] for i in (self.y_labels[j] for j in valid_indices)])
+        y_lower = np.array([self.cat_row["Unc_" + i][0] for i in (self.y_labels[j] for j in valid_indices)])
+        y_upper = np.array([self.cat_row["Unc_" + i][1] for i in (self.y_labels[j] for j in valid_indices)])
+
+        y_lower = y_vals + y_lower
+        y_upper = y_vals + y_upper
+
+        x_vals = [self.x_cens[i].value for i in valid_indices]
+        bin_edges1 =[-(self.x_bins_edges[i] - self.x_cens[i]).value for i in valid_indices]
+        bin_edges2 = [(self.x_bins_edges[i+1] - self.x_cens[i]).value for i in valid_indices]
+
+        y_vals = [y_vals[i] / x_vals[i] for i in range(0, np.size(y_vals))]
+        y_upper = [y_upper[i] / x_vals[i] for i in range(0, np.size(y_vals))]
+        y_lower = [y_lower[i] / x_vals[i] for i in range(0, np.size(y_vals))]
+
+        y_cens = np.array([gmean([y_lower[i], y_upper[i]]) for i in range(0, np.size(y_lower))])
+
+        y_upper = np.array([y_upper[i] - y_vals[i] for i in range(0, np.size(y_lower))])
+        y_lower = np.array([y_vals[i] - y_lower[i] for i in range(0, np.size(y_lower))])
+
+        ax.loglog()
+
+        ax.errorbar(x_vals, y_vals,
+                    yerr=(y_lower, y_upper),
+                    elinewidth=1, linewidth=0, color='black')
+
+        # Place the x-axis uncertainties in the center of the y-axis uncertainties.
+        ax.errorbar(x_vals, y_cens,
+                    xerr=(bin_edges1, bin_edges2),
+                    elinewidth=1, linewidth=0, color='black')
+
+        x_model = np.logspace(np.log10(min(x_vals)), np.log10(max(x_vals)), 25)
+
+        if self.spec_type == "PowerLaw":
+
+            y_model = PowerLaw1D(amplitude=self.int_flux,
+                                 x_0=self.pivot_en,
+                                 alpha=self.spec_index)
+
+        elif self.spec_type == "LogParabola":
+
+            y_model = LogParabola1D(amplitude=self.int_flux,
+                                    x_0=self.pivot_en,
+                                    alpha=self.spec_index,
+                                    beta = self.beta)
+
+        elif self.spec_type == "PLSuperExpCutoff":
+            y_model = ExponentialCutoffPowerLaw1D(amplitude=self.int_flux,
+                                                  x_0=self.pivot_en,
+                                                  alpha=self.spec_index,
+                                                  x_cutoff = self.cutoff)
+
+        ax.plot(x_model, y_model(x_model))
+
+        return ax
```

Comment:
These four values for `SpectrumType` do actually occur in the 3FHL catalog:
```python
>>> from astropy.table import Table
>>> table = Table.read('http://fermi.gsfc.nasa.gov/ssc/data/access/lat/4yr_catalog/gll_psc_v16.fit')
>>> set(table['SpectrumType'])
{'PowerLaw        ', 'LogParabola     ', 'PLExpCutoff     ', 'PLSuperExpCutoff'}
```
I think the `PLSuperExpCutoff` spectral model isn't implemented anywhere right now?
If so, maybe just raise a ValueError if the user requests a source with such a spectrum here,
and leave a TODO marker or file a new Github issue that this should be implemented?

Do you mean you're also getting an incorrect image for LogParabola and PLExpCutoff?
Do you have an example source what we can use for testing, i.e. look at the expected image from the PDFs published by Fermi and the one you get from Gammapy?